WP_Router_Page.class.php

```diff
@@ -122,17 +122,19 @@ public function edit_query( WP_Query $query ) {
 		}
 	}
 
-	/**
-	 * Override the global $pages array to yield our content
-	 *
-	 * @param object $post
-	 * @return void
-	 */
-	public function set_post_contents( $post ) {
-		global $pages;
-		$pages = array($this->contents);
-		// TODO: add a facility for multi-page documents?
-	}
+    /**
+     * Override the global $pages array to yield our content
+     *
+     * @param object $post
+     *
+     * @return void
+     */
+    public function set_post_contents( $post ) {
+        if ( WP_Router_Page::POST_TYPE == $post->post_type ) {
```

Comment:
Should we have a strict comparison?

Also, we tend apply the logic of "bailing early" in our codebase. Maybe we can invert the conditional and return. And if not, execute the code.